data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/meter/KafkaConnectMetrics.java

```diff
@@ -0,0 +1,162 @@
+        this(pluginMetrics, emptyList());
+    }
+
+    public KafkaConnectMetrics(final PluginMetrics pluginMetrics,
+                               final Iterable<Tag> tags) {
+        this(pluginMetrics, getMBeanServer(), tags);
+    }
+
+    public KafkaConnectMetrics(final PluginMetrics pluginMetrics,
+                               final MBeanServer mBeanServer,
+                               final Iterable<Tag> tags) {
+        this.pluginMetrics = pluginMetrics;
+        this.mBeanServer = mBeanServer;
+        this.tags = tags;
+    }
+
+    private static MBeanServer getMBeanServer() {
+        List<MBeanServer> mBeanServers = MBeanServerFactory.findMBeanServer(null);
+        if (!mBeanServers.isEmpty()) {
+            return mBeanServers.get(0);
+        }
+        return ManagementFactory.getPlatformMBeanServer();
+    }
+
+    private static String sanitize(String value) {
+        return value.replaceAll("-", ".");
+    }
+
+    public void bindConnectMetrics() {
+        registerMetricsEventually(CONNECT_WORKER_METRICS_NAME, (o, tags) -> {
+            CONNECT_WORKER_METRICS_LIST.forEach(
+                    (metricName) -> registerFunctionGaugeForObject(o, metricName, tags)
+            );
+            return null;
+        });
+    }
+
+    public void bindConnectorMetrics() {
+        registerMetricsEventually(SOURCE_TASK_METRICS_NAME, (o, tags) -> {
+            SOURCE_TASK_METRICS_LIST.forEach(
+                    (metricName) -> registerFunctionGaugeForObject(o, metricName, tags)
+            );
+            return null;
+        });
+    }
+
+    private void registerMetricsEventually(String type,
+                                           BiFunction<ObjectName, Tags, Void> perObject) {
+        try {
+            Set<ObjectName> objs = mBeanServer.queryNames(new ObjectName(JMX_DOMAIN + ":type=" + type + ",*"), null);
+            if (!objs.isEmpty()) {
+                for (ObjectName o : objs) {
+                    perObject.apply(o, Tags.concat(tags, nameTag(o)));
+                }
+            }
+        } catch (MalformedObjectNameException e) {
+            throw new RuntimeException("Error registering Kafka Connect JMX based metrics", e);
+        }
+    }
+
+    private Iterable<Tag> nameTag(ObjectName name) {
+        Tags tags = Tags.empty();
+
+        String clientId = name.getKeyProperty("client-id");
+        if (clientId != null) {
+            tags = Tags.concat(tags, "client.id", clientId);
+        }
+
+        String nodeId = name.getKeyProperty("node-id");
+        if (nodeId != null) {
+            tags = Tags.concat(tags, "node.id", nodeId);
+        }
+
+        String connectorName = name.getKeyProperty("connector");
+        if (connectorName != null) {
+            tags = Tags.concat(tags, "connector", connectorName);
+        }
+
+        String taskName = name.getKeyProperty("task");
+        if (taskName != null) {
+            tags = Tags.concat(tags, "task", taskName);
```

Comment:
nit: more string in this method to constant.

Reply:
ok